server/util-server.js

```diff
@@ -203,6 +204,29 @@ exports.dnsResolve = function (hostname, resolverServer, rrtype) {
     });
 };
 
+/**
+ * Run a query on SQL Server
+ * @param {string} connectionString The database connection string
+ * @param {string} query The query to validate the database with
+ * @returns {Promise<(string[]|Object[]|Object)>}
+ */
+exports.sqlserver = function (connectionString, query) {
+    return new Promise((resolve, reject) => {
+        sql.on("error", err => {
+            reject(err);
+        });
+
+        sql.connect(connectionString).then(pool => {
+            return pool.request()
+                .query(query);
+        }).then(result => {
+            resolve(result);
+        }).catch(err => {
+            reject(err);
+        });
```

Comment:
I don't know if it's necessary, but wouldn't it be good to close the database?

See. https://www.npmjs.com/package/mssql#close

Reply:
Its a good question. The connections seem to be handled correctly, a connection pool is left existing as sleeping. I think this is the correct way for it to work.

From the docs they say not to close the pool: (section "The Global Connection Pool")

![image](https://user-images.githubusercontent.com/17788706/168285766-b431cdc4-b010-4efb-9710-85395e485210.png)


I did a test of this, I set the monitor to run ever 20 seconds and then monitored the connections on the db:
![image](https://user-images.githubusercontent.com/17788706/168284868-e912e238-077e-4d68-ac3c-495103aae2d8.png)

Here's the db after running for a few minutes:
![image](https://user-images.githubusercontent.com/17788706/168285039-b1af0baa-1a12-48f7-9eb3-96aa875b9f16.png)

After disabling the monitor, in about 60 seconds, sql server drops the pool:
![image](https://user-images.githubusercontent.com/17788706/168285476-1209d557-f744-439b-aa52-1e86db8021e7.png)

Reply:
![image](https://user-images.githubusercontent.com/59953812/168289508-40777ef4-fccd-408b-bdf1-1bd059fb54b7.png)

It really sounds like an example of a script / CRON job.

In essence, each check calls for a connection to the MSSQL server and sends a query. In essence, leaving the connection after this step is unnecessary because the code for the next connection is called anyway.

The documentation says:
*`sql.connect()` will return the existing global pool if it exists or create a new one if it doesn't*
,but if the monitor performs a check every hour, for example, it is unnecessary to leave the connection open. Personally, I think it could unnecessarily burden the line unreasonably, and it would be better to call the `.close()` method.

Will anything improve / worsen when closing the connection manually after performing the test?

Reply:
Hmm sql server automatically closes them out after they go to sleep for a minute, so it seems like an extra step. There is no overflow of connections and its a process sql server has built in.

With the pool it isn't created a new connection each time, but recycling the previous one. W/ the default monitor set at 60 seconds the old connection will be reused before sql removes it. If your time is longer than that, sql will auto clean it. 

For monitors running inside the window, performance will be worse (imperceptible probably :D unless you are checking a million databases at once?) if we close the pool. If monitors are longer than the sql server window, there will be no change in performance as sql will have automatically closed the pool by that time.

The db garbage collection processes work nicely.

Here's a link to a boring read: https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/sql-server-connection-pooling. It seems like as is is the best way. The loss in performance in short monitors is minimal, so its fine if we add it in. :) I'm not sure how to go about it w/ the promise structure though.

From reading the [code](https://www.npmjs.com/package/mssql), it seems that if we really want to close the pool manually, we have to call another connection, then call close. (Advanced Pool Management section), which probably adds more overhead than letting sql server garbage collection do its job. The pool is different than making lots of sql.connect's.

```
sql.connect(connectionString).then(pool => {
            return pool.close();
        }).catch(err => {
            reject(err);
        });
```

Reply:
I thought it could be done like this:
```js
sql.connect(connectionString).then(pool => {
    const queryResult = pool.request().query(query);
    pool.close();

    return queryResult;
}).then(result => {
    resolve(result);
}).catch(err => {
    reject(err);
});
```

\
The question is whether each MSSQL server closes the connection itself. What if it's set to hold the connection?

I don't really know if to close the database or not, do it the way you think is the best.
If @louislam wants to close the database, then you can add it there in the future.

Reply:
Thanks! I tried that, but it was not happy :) The query is not actually run until it is resolved, and by then the pool had already been killed. The pooling exists in the three supported releases of sql server, so we are good there. If an issues comes up w/ pooling for something unsupported by MS, it should be taken care of at the `mssql` package level. Of course, you can override the default settings in your sql server install if you want it to do something different. I think thats beyond the scope here though. Thanks for reviewing!